openfisca_uk_data/datasets/frs/frs.py

```diff
@@ -1,87 +1,635 @@
+    ) * 52
+
+    # Add self-employed income (correcting one person in 2018)
+    seincamt = (
+        (job.SEINCAMT.groupby(job.person_id).sum())
+        .reindex(person.index)
+        .fillna(0)
+    )
+    frs["self_employment_income"] = (
+        np.where(
+            (year == 2018) & (person.index == 806911),
+            seincamt,
+            person.SEINCAM2,
+        )
+        * 52
+    )
+
+    frs["tax_free_savings_income"] = (
+        account.ACCINT[account.ACCOUNT == 21]
+        .groupby(account.person_id)
+        .sum()
+        .reindex(index=person.index)
+        .fillna(0)
+        * 52
+    )
+    frs["savings_interest_income"] = (
+        (account.ACCINT * np.where(account.ACCTAX == 1, 1.25, 1))[
+            account.ACCOUNT.isin((1, 3, 5, 27, 28))
+        ]
+    ).groupby(account.person_id).sum().reindex(index=person.index).fillna(
+        0
+    ) * 52 + frs[
+        "tax_free_savings_income"
+    ][
+        ...
+    ]
+    frs["dividend_income"] = (
+        account.ACCINT * np.where(account.INVTAX == 1, 1.25, 1)
+    )[
+        ((account.ACCOUNT == 6) & (account.INVTAX == 1))
+        | account.ACCOUNT.isin((7, 8))
+    ].groupby(
+        account.person_id
+    ).sum().reindex(
+        index=person.index
+    ).fillna(
+        0
+    ) * 52
+    is_head = person.HRPID == 1
+    frs["property_income"] = (
+        is_head
+        * (
+            pd.Series(
+                (household.TENTYP2.isin((5, 6)) * household.SUBRENT)[
+                    person.household_id
+                ].values,
+                index=person.index,
+            ).fillna(0)
+        )
+        + person.CVPAY
+        + person.ROYYR1
+    ) * 52
+
+    # Discrepancy in maintenance income (UKMOD appears to use last amount rather than usual, with "not usual" answer):
+
+    INVERTED_USUAL_AMOUNT_HOUSEHOLDS = (
+        1458600,
+        1033500,
+        322100,
+        42100,
+        590400,
+        645300,
+        1901400,
+        388000,
+    )
+
+    frs["maintenance_income"] = (
+        max_(
+            pd.Series(
+                where(person.MNTUS1 == 2, person.MNTUSAM1, person.MNTAMT1)
+            ).fillna(0),
+            0,
+        )
+        + max_(
+            pd.Series(
+                where(
+                    (person.MNTUS2 == 2)
+                    & ~(
+                        (year == 2018)
+                        & (
+                            person.household_id.isin(
+                                INVERTED_USUAL_AMOUNT_HOUSEHOLDS
+                            )
+                        )
+                    ),
+                    person.MNTUSAM2,
+                    person.MNTAMT2,
+                )
+            ).fillna(0),
+            0,
+        )
+    ) * 52
+
+    frs["miscellaneous_income"] = (
+        oddjob.OJAMT[oddjob.OJNOW == 1]
+        .groupby(oddjob.person_id)
+        .sum()
+        .reindex(person.index)
+        .fillna(0)
+        + person.ALLPAY2
```

Comment:
```suggestion
        + person[["ALLPAY2", "ROYYR2", "ROYYR3", "CHAMTERN", "CHAMTTST"]].sum(axis=1)
```
could also name those columns as a constant first